perfmetrics/scripts/ls_metrics/listing_benchmark.py

```diff
@@ -0,0 +1,473 @@
+  """
+  for curr_package in packages:
+    log.info('Checking whether %s is installed.\n', curr_package)
+
+    exit_code = subprocess.call(
+        '{} --version'.format(curr_package), shell=True)
+    print('')
+
+    if exit_code != 0:
+      log.error('%s not installed. Please install. Aborting!\n', curr_package)
+      subprocess.call('bash', shell=True)
+
+  return
+
+
+if __name__ == '__main__':
+  argv = sys.argv
+  if len(argv) < 3:
+    raise TypeError('Incorrect number of arguments.\n'
+                    'Usage: '
+                    'python3 listing_benchmark.py [--keep_files] [--upload] [--num_samples NUM_SAMPLES] [--message MESSAGE] --command COMMAND config_file')
+
+  args = ParseArguments(argv)
+
+  CheckDependencies(['gsutil', 'gcsfuse'])
+
+  f = open(os.path.abspath(args.config_file))
+  config_json = json.load(f)
+  f.close()
+
+  directory_structure = ParseDict(config_json, directory_proto.Directory())
+
+  gcs_bucket = MountGCSBucket(directory_structure.name)
+
+  log.info('Started checking the directory structure in the bucket.\n')
+  directory_structure_present = CompareDirectoryStructure('gs://{}/'.format(directory_structure.name),
+                                                                            directory_structure)
+  if not directory_structure_present:
+    log.info('Similar directory structure not found in the GCS bucket. Creating a new one.\n')
+  else:
+    log.info('Similar directory already present in the bucket. Not making a new one.\n')
+
+  # Removing the already present folder in persistent disk so as to create the
+  # files from scratch.
+  persistent_disk = 'persistent_disk'
+  if os.path.exists('./{}'.format(persistent_disk)):
+    subprocess.call('rm -rf {}'.format(persistent_disk), shell=True)
+
+  # If similar directory structure not found in the GCS bucket then delete all
+  # the files in the bucket and make it from scratch.
+  if not directory_structure_present:
+    log.info('Deleting previously present directories in the GCS bucket.\n')
+    subprocess.call('gsutil -m rm -r gs://{}/*'.format(directory_structure.name),
+                    shell=True, stdout = subprocess.DEVNULL, stderr = subprocess.STDOUT)
+
+  # Creating a temp directory which will be needed by the generate_files method to
+  # create files in batches.
+  temp_dir = generate_files.TEMPORARY_DIRECTORY
+  if os.path.exists(os.path.dirname(temp_dir)):
+    subprocess.call('rm -rf {}'.format(os.path.dirname(temp_dir)), shell=True)
+  subprocess.call('mkdir -p {}'.format(temp_dir), shell=True)
+
+  exit_code = CreateDirectoryStructure('gs://{}/'.format(directory_structure.name), './{}/'.format(persistent_disk),
+                                       directory_structure, not directory_structure_present)
+
+  # Deleting the temp folder after the creation of files is done.
+  subprocess.call('rm -rf {}'.format(os.path.dirname(temp_dir)) ,shell=True)
+
+  if exit_code != 0:
+    UnmountGCSBucket(gcs_bucket)
```

Comment:
ideally mounting of gcsfuse should happen after creating the files. if exit_code == 0, then do mount